test/built-ins/Temporal/Duration/from/duration.js

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.duration.from
+description: Temporal.Duration.from will crate a Duration from a given Duration.
+info: |
+  1. If Type(item) is Object and item has an [[InitializedTemporalDuration]] internal slot, then
+  a. Return ? CreateTemporalDuration(item.[[Years]], item.[[Months]], item.[[Weeks]], item.[[Days]], item.[[Hours]], item.[[Minutes]], item.[[Seconds]], item.[[Milliseconds]], item.[[Microseconds]], item.[[Nanoseconds]]).
+  2. Return ? ToTemporalDuration(item).
+features: [Temporal]
+---*/
+function assertDuration(duration, years, months, weeks, days, hours,
+    minutes, seconds, milliseconds, microseconds, nanoseconds, sign, blank) {
+  assert.sameValue(years, duration.years, "years");
+  assert.sameValue(months, duration.months, "months");
+  assert.sameValue(weeks, duration.weeks, "weeks");
+  assert.sameValue(days, duration.days, "days");
+  assert.sameValue(hours, duration.hours, "hours");
+  assert.sameValue(minutes, duration.minutes, "minutes");
+  assert.sameValue(seconds, duration.seconds, "seconds");
+  assert.sameValue(milliseconds, duration.milliseconds, "milliseconds");
+  assert.sameValue(microseconds, duration.microseconds, "microseconds");
+  assert.sameValue(nanoseconds, duration.nanoseconds, "nanoseconds");
+  assert.sameValue(sign, duration.sign, "sign");
+  assert.sameValue(blank, duration.blank, "blank");
+}
```

Comment:
If this is going to be used many times, please add it to https://github.com/tc39/test262/blob/main/harness/temporalHelpers.js